src/shogun/machine/TypedMachine.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Shubham Shukla
+ */
+
+#ifndef _TYPEDMACHINE_H__
+#define _TYPEDMACHINE_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/base/progress.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/lib/common.h>
+#include <shogun/machine/LinearMachine.h>
+
+namespace shogun
+{
+	class CFeatures;
+	class CLabels;
+	
+	template <class P, class T>
+	class CTypedMachine : public T
```

Comment:
i think the name is not the best. I know we used this in the gist, but maybe we can call this FeatureDispatcherMachine (or something better)

Reply:
yes i agree

Reply:
CMachineFeatureTypeCRTP would be more descriptive, but also more ugly ;)